From 707c827c4de8ba40798e4216f6614644af4a5f98 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Fri, 24 Jan 2025 14:52:02 +0100
Subject: [PATCH] Fix build with liburcu-0.15.0

:Release Notes:

:Detailed Notes:
Fixes:
http://gecko.lge.com:8000/Errors/Details/1096867

In file included from TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/recipe-sysroot/usr/include/lttng/tracepoint-rcu.h:10,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/recipe-sysroot/usr/include/lttng/tracepoint.h:13,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/git/modules/weboscompositor/pmtrace_surfacemanager_provider.h:30,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/git/modules/weboscompositor/weboscompositortracer.h:22,
                 from TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/git/modules/weboscompositor/weboswindowmodel.cpp:20:
TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/recipe-sysroot/usr/include/urcu/compiler.h:205:1: error: template with C linkage
  205 | template <typename T>
      | ^~~~~~~~
TOPDIR/BUILD/work/qemux86_64-webos-linux/luna-surfacemanager/2.0.0-420/git/modules/weboscompositor/pmtrace_surfacemanager_provider.h:24:1: note: 'extern "C"' linkage started here
   24 | extern "C"{
      | ^~~~~~~~~~

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-17603] Fix build with liburcu-0.15.0

Change-Id: Iaef58b3ae8dffa7a14353935b3baec52d11532e0
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/luna-surfacemanager/+/466650 Fix build with liburcu-0.15.0]

 .../pmtrace_surfacemanager_provider.h                | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/modules/weboscompositor/pmtrace_surfacemanager_provider.h b/modules/weboscompositor/pmtrace_surfacemanager_provider.h
index 11c4474c..1676f091 100644
--- a/modules/weboscompositor/pmtrace_surfacemanager_provider.h
+++ b/modules/weboscompositor/pmtrace_surfacemanager_provider.h
@@ -20,15 +20,15 @@
 #undef TRACEPOINT_INCLUDE
 #define TRACEPOINT_INCLUDE "./pmtrace_surfacemanager_provider.h"
 
-#ifdef __cplusplus
-extern "C"{
-#endif /*__cplusplus */
-
 #if !defined(PMTRACE_SURFACEMANAGER_PROVIDER_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
 #define PMTRACE_SURFACEMANAGER_PROVIDER_H
 
 #include <lttng/tracepoint.h>
 
+#ifdef __cplusplus
+extern "C"{
+#endif /*__cplusplus */
+
 /* "message" tracepoint should be used for single event trace points */
 TRACEPOINT_EVENT(
     pmtrace_surfacemanager,
@@ -101,8 +101,8 @@ TRACEPOINT_EVENT(
     TP_ARGS(const char*, text),
     TP_FIELDS(ctf_string(scope, text)))
 
-#endif /* PMTRACE_SURFACEMANAGER_PROVIDER_H */
-
 #ifdef __cplusplus
 }
 #endif /*__cplusplus */
+
+#endif /* PMTRACE_SURFACEMANAGER_PROVIDER_H */
